src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -802,19 +802,8 @@ internal override bool ApplyNullableTransforms(byte defaultTransformFlag, Immuta
                 }
             }
 
-            if (!haveChanges)
-            {
-                allTypeArguments.Free();
-                result = this;
-            }
-            else
-            {
-                TypeMap substitution = new TypeMap(this.OriginalDefinition.GetAllTypeParameters(),
-                                                   allTypeArguments.ToImmutableAndFree());
-
-                result = substitution.SubstituteNamedType(this.OriginalDefinition).WithTupleDataFrom(this);
-            }
-
+            result = haveChanges ? this.WithTypeArguments(allTypeArguments.ToImmutable()) : this;
```

Comment:
Is this just a nullable refactor (no behavior changes)?

Reply:
Yes, this is a just a refactor to extract `WithTypeArguments()`.

---
In reply to: [373269221](https://github.com/dotnet/roslyn/pull/41160#discussion_r373269221) [](ancestors = 373269221)